var/spack/repos/builtin/packages/apple-clang/package.py

```diff
@@ -2,57 +2,23 @@
 # Spack Project Developers. See the top-level COPYRIGHT file for details.
 #
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
-import re
-
 from spack.package import *
 
 
-class AppleClang(BundlePackage):
+class AppleClang(BundlePackage, CompilerPackage):
     """Apple's Clang compiler"""
 
     homepage = "https://developer.apple.com/videos/developer-tools/compiler-and-llvm"
     has_code = False
 
     maintainers("alalazo")
 
-    executables = ["^clang$", r"^clang\+\+$", "^ld.lld$", "^lldb$"]
-
-    @classmethod
-    def determine_version(cls, exe):
-        version_regex = re.compile(
-            # Apple's LLVM compiler has its own versions, which are
-            # different from vanilla LLVM
-            r"^Apple (?:LLVM|clang) version ([^ )]+)",
-            # Multi-line, since 'Apple clang' may not be on the first line
-            # in particular, when run as gcc, it seems to output
-            # "Configured with: --prefix=..." as the first line
-            re.M,
-        )
-        try:
-            compiler = Executable(exe)
-            output = compiler("--version", output=str, error=str)
-            match = version_regex.search(output)
-            if match:
-                return match.group(match.lastindex)
-        except Exception:
-            pass
-
-        return None
-
-    @classmethod
-    def determine_variants(cls, exes, version_str):
-        compilers = {}
-        for exe in exes:
-            if "clang++" in exe:
-                compilers["cxx"] = exe
-            elif "clang" in exe:
-                compilers["c"] = exe
-            elif "ld.lld" in exe:
-                compilers["ld"] = exe
-            elif "lldb" in exe:
-                compilers["lldb"] = exe
+    compiler_languages = ["c", "cxx"]
+    c_names = ["clang"]
+    cxx_names = ["clang++"]
 
-        return "", {"compilers": compilers}
+    version_regex = r"^Apple (?:LLVM|clang) version ([^ )]+)"
```

Comment:
Here we are missing the multiline option. Not sure it matters for recent Apple Clang.

Reply:
Is there any reason to keep the `^`? If we get rid of that then multiline becomes completely irrelevant (although it's also irrelevant for every apple-clang version output I've been able to compare to).

Reply:
Tests are passing, so let's close this discussion. We can get back to it if we see faulty detection.